table/block_based/full_filter_block.cc

```diff
@@ -254,15 +255,28 @@ void FullFilterBlockReader::MayMatch(
   // expensive compared to autovector
   Slice* keys[MultiGetContext::MAX_BATCH_SIZE];
   bool may_match[MultiGetContext::MAX_BATCH_SIZE] = {false};
```

Comment:
Shouldn't these may_match flags default to true? I.e. if a key is not in the prefix extractor's domain, we should treat it as potentially present, no? (Edit: this probably does not make a practical difference though since the code only seems to use the first num_keys elements of this array from this point on, and those get set by FullFilterBitsReader. We might still consider changing it for the sake of consistency/clarity.)

Reply:
Yes, makes sense to be conservative.